org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/IProblem.java

```diff
@@ -2553,6 +2553,22 @@ public interface IProblem {
 	 * @noreference preview feature
 	 */
 	int CannotInferRecordPatternTypes = PreviewRelated + 1940;
+	
+	/**
+	 * @since 3.36
+	 * @noreference preview feature
+	 */
+	int unnamedIsMissingMainMethod = PreviewRelated + 1942;
+	/**
+	 * @since 3.36
+	 * @noreference preview feature
+	 */
+	int unnamedRequiresJava21 = PreviewRelated + 1943;
```

Comment:
There is a dedicated error message that exists for all features that require preview to be enabled, and if you use that feature, there is already a quick fix set up to enable preview. I think it might be easier to use the existing error message.

See https://github.com/eclipse-jdt/eclipse.jdt.core/blob/3d920573614a988a92901edfa33d0fa0796411af/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/impl/JavaFeature.java#L72 and `ProblemReporter.validateJavaFeatureSupport`

Reply:
Thanks, I've fixed it in latest version of the PR.